tests/core/unittests/test_experiment.py

```diff
@@ -0,0 +1,423 @@
+        new_config['status'] = 'pending'
+        new_config.pop('_id')
+        new_config.pop('something_to_be_ignored')
+        assert found_config[0] == new_config
+        assert exp.name == new_config['name']
+        assert exp.refers is None
+        assert exp.metadata == new_config['metadata']
+        assert exp.pool_size == new_config['pool_size']
+        assert exp.max_trials == new_config['max_trials']
+        assert exp.status == new_config['status']
+        #  assert exp.algorithms == new_config['algorithms']
+
+    def test_inconsistent_1_set_before_init_no_hit(self, random_dt, new_config):
+        """Test inconsistent configuration because of name."""
+        exp = Experiment(new_config['name'])
+        new_config['name'] = 'asdfaa'
+        with pytest.raises(ValueError) as exc_info:
+            exp.configure(new_config)
+        assert 'inconsistent' in str(exc_info.value)
+
+    def test_inconsistent_2_set_before_init_no_hit(self, random_dt, new_config):
+        """Test inconsistent configuration because of user."""
+        exp = Experiment(new_config['name'])
+        new_config['metadata']['user'] = 'asdfaa'
+        with pytest.raises(ValueError) as exc_info:
+            exp.configure(new_config)
+        assert 'inconsistent' in str(exc_info.value)
+
+    def test_inconsistent_3_set_before_init_no_hit(self, random_dt, new_config):
+        """Test inconsistent configuration because of datetime."""
+        exp = Experiment(new_config['name'])
+        new_config['metadata']['datetime'] = 123
+        with pytest.raises(ValueError) as exc_info:
+            exp.configure(new_config)
+        assert 'inconsistent' in str(exc_info.value)
+
+    def test_get_after_init_plus_hit_no_diffs(self, exp_config):
+        """Return a configuration dict according to an experiment object.
+
+        Before initialization is done, it can be the case that the pair (`name`,
+        user's name) has not hit the database. return a yaml compliant form
+        of current state, to be used with :mod:`metaopt.resolve_config`.
+        """
+        exp = Experiment('supernaedo2')
+        # Deliver an external configuration to finalize init
+        exp.configure(exp_config[0][1])
+        assert exp._init_done is True
+        exp_config[0][1]['status'] = 'pending'
+        assert exp.configuration == exp_config[0][1]
+        exp_config[0][1]['status'] = 'broken'  # don't ruin resource
+
+    def test_try_set_after_init(self, exp_config):
+        """Cannot set a configuration after init (currently)."""
+        exp = Experiment('supernaedo2')
+        # Deliver an external configuration to finalize init
+        exp.configure(exp_config[0][1])
+        assert exp._init_done is True
+        with pytest.raises(RuntimeError) as exc_info:
+            exp.configure(exp_config[0][1])
+        assert 'cannot reset' in str(exc_info.value)
+
+    def test_after_init_algorithms_are_objects(self, exp_config):
```

Comment:
Maybe it would be better to add explicit skips to make sure we do not forget to implement those tests when time comes.